src/pudl/transform/eia861.py

```diff
@@ -818,5 +845,9 @@ def transform(raw_dfs, eia861_tables=pc.pudl_tables["eia861"]):
             logger.info(f"Transforming raw EIA 861 DataFrames for {table} "
                         f"concatenated across all years.")
             eia861_transform_functions[table](raw_dfs, tfr_dfs)
-
+    # for each transformed table, convert year col to date
+    for table in tfr_dfs:
+        tfr_dfs[table] = (
+            tfr_dfs[table].pipe(pudl.helpers.convert_to_date)
+        )
```

Comment:
I think we probably want to apply this conversion **before** all of the transform functions are run, rather than after, so that any transform functions which need to operate on the date column has a date to work with. In general we're trying to get any column re-names or broad brush changes made as early in the process as possible, so that as much of the code as possible can work with the data in a uniform way.

In the `pudl.transform.ferc714` module, I did this by creating a `_pre_transform()` function which pre-populated the `tfr_dfs` dictionary with only slightly transformed raw dataframes, inside the main loop of the `transform()` function and then all of the transform functions operated only on those transformed dataframes.

Reply:
Good idea; I'll make something similar.